components/nif/nifstream.cpp

```diff
@@ -24,4 +24,21 @@ namespace Nif
         t.scale = getFloat();
         return t;
     }
+
+    ///Currently specific for 4.0.0.2 and earlier
+    bool NIFStream::getBoolean()
+    {
+        return getInt() != 0;
+    }
+
+    ///Read in a string, either from the string table using the index (currently absent) or from the stream using the specified length
+    std::string NIFStream::getString()
+    {
+        return getSizedString();
+    }
+
+    // Convenience utility functions: get the versions of the currently read file
+    unsigned int NIFStream::getVersion() { return file->getVersion(); }
+    unsigned int NIFStream::getUserVersion() { return file->getBethVersion(); }
```

Comment:
Should this be? Otherwise what's the point of this function?
```c++
unsigned int NIFStream::getUserVersion() { return file->getUserVersion(); }
```

Reply:
You can tell the intention by the existence of userVer/getUserVersion in NIFFile


I guess there accidentally was never a case in testing where the user version mattered and most files worked by accident.